lib/utils/startUnixSocket.js

```diff
@@ -0,0 +1,58 @@
+'use strict';
+
+const fs = require('fs');
+const net = require('net');
+
+function startUnixSocket(listeningApp, socket, cb) {
+  const chmodSocket = (done) => {
+    // chmod 666 (rw rw rw) - octal
+    const READ_WRITE = 438;
+    fs.chmod(socket, READ_WRITE, done);
+  };
+
+  const startSocket = () => {
+    listeningApp.on('error', (err) => {
+      cb(err);
+    });
+
+    // 511 is the default value for the server.listen backlog parameter
+    // https://nodejs.org/api/net.html#net_server_listen
+    listeningApp.listen(socket, 511, (err) => {
+      if (err) {
+        cb(err);
+      } else {
+        chmodSocket(cb);
+      }
+    });
+  };
+
+  fs.access(socket, fs.constants.F_OK, (e) => {
```

Comment:
We don't need to use a callback. Just use `promisify`.

Reply:
@hiroppy is this what you meant?:

https://github.com/webpack/webpack-dev-server/pull/2061/files#diff-2e3826007762dc4188ae8303ccd8b036R32-R38

Reply:
Yes, thanks!